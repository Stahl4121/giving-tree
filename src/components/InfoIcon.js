import React, { useEffect } from 'react';

import { Info } from '@material-ui/icons';
import IconButton from '@material-ui/core/IconButton';
import Popover from '@material-ui/core/Popover';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  button: {
    cursor: 'pointer',
  },
  typography: {
    maxWidth: 300,
    padding: theme.spacing(2),
  },
}));

export default function ContactPopout(props) {
  const classes = useStyles();
  const [anchorEl, setAnchorEl] = React.useState(null);

  const handleClick = event => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  const open = Boolean(anchorEl);
  const id = open ? 'info-popover' : undefined;

  return (
    <div>
      <IconButton className={classes.button} aria-describedby={id} variant="contained" color="primary" onClick={handleClick} variant='outlined' color='secondary'>
        <Info />
      </IconButton>
      <Popover
        id={id}
        open={open}
        anchorEl={anchorEl}
        onClose={handleClose}
        anchorOrigin={{
          vertical: 'top',
          horizontal: 'right',
        }}
        transformOrigin={{
          vertical: 'bottom',
          horizontal: 'left',
        }}
      >
        <Typography
          className={classes.typography}
          variant="body2">
          {props.infoMessage}
        </Typography>
      </Popover>
    </div>
  );
}