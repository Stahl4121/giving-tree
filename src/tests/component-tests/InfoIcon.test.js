import React from "react";
import { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import InfoIcon from "../../components/InfoIcon.js";

describe('Component: InfoIcon', () => {
  var enzyme = require('enzyme');
  enzyme.configure({ adapter: new Adapter() });

  it('Renders correctly', () => {
    const message = "";
    const wrap = enzyme.mount(<InfoIcon infoMessage={message}/>)
    expect(wrap).toMatchSnapshot();
  });

})
