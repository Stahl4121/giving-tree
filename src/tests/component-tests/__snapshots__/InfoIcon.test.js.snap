// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: InfoIcon Renders correctly 1`] = `ReactWrapper {}`;
